src/tpm2_wrap.c

```diff
@@ -2665,6 +2744,30 @@ int wolfTPM2_NVReadAuth(WOLFTPM2_DEV* dev, WOLFTPM2_NV* nv,
         wolfTPM2_SetAuthHandle(dev, 0, &nv->handle);
     }
 
+    /* Read the NV Index publicArea to have up to date NV Index Name */
+    rc = wolfTPM2_NVReadPublic(dev, nv->handle.hndl, &nvPublic);
+    if (rc != TPM_RC_SUCCESS) {
+    #ifdef DEBUG_WOLFTPM
+        printf("Failed to read fresh NvPublic\n");
+    #endif
+        return TPM_RC_FAILURE;
+    }
+
+    /* Compute NV Index name in case of parameter encryption */
+    rc = TPM2_HashNvPublic(&nvPublic, (byte*)&nv->handle.name.name,
+                            &nv->handle.name.size);
+    if (rc != TPM_RC_SUCCESS) {
+        return rc;
+    }
+
+    /* Necessary, because NVWrite has two handles, second is NV Index */
+    rc = wolfTPM2_SetNameHandle(dev, 0, &nv->handle);
+    rc |= wolfTPM2_SetNameHandle(dev, 1, &nv->handle);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("Storing NV Index Name failed\n");
```

Comment:
Please wrap this printf in `DEBUG_WOLFTPM`. The `rc` return code will not be valid with your above or'ing of the return code. Consider just returning `TPM_RC_FAILURE`.

Reply:
good point. Thank you